src/pages/account.jsx

```diff
@@ -0,0 +1,218 @@
+import Card from "@/components/Card";
+import userpfpplaceholder from "../../public/userpfpplaceholder.svg";
+
+import Image from "next/image";
+import { useRouter } from "next/router";
+
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+
+import { PencilSquareIcon } from "@heroicons/react/24/outline";
+import {
+  ExclamationTriangleIcon,
+  ExclamationCircleIcon,
+} from "@heroicons/react/24/solid";
+import { consts } from "@/utils/consts";
+
+/**
+ * User account management page
+ *
+ * @returns {React.ReactElement} The user account management page
+ */
+export default function Account() {
+  const [data, setData] = useState();
+  const [editName, setEditName] = useState(false);
+  const [name, setName] = useState("");
+  const [showDeactivateModal, setShowDeactivateModal] = useState(false);
+
+  const router = useRouter();
+  // TODO: replace with session info
+  const placeholderUser = "65358b82eccc98bd5877fcbb";
+
+  useEffect(() => {
+    fetch(`/api/users/${placeholderUser}`)
+      .then((res) => res.json())
+      .then((data) => setData(data));
+  }, [router.query]);
+
+  if (!data || data === undefined || !data.success) {
+    return <div>loading</div>;
+  }
+
+  const user = data.data;
+
+  return (
+    <Card cardStyle="mt-16 min-w-fit">
+      {showDeactivateModal ? (
```

Comment:
possibly break the modal into its own component. You don't need to make a separate file for it though. Can keep it in this file

Reply:
done!